### PR TITLE
Add cherrypick.sh script

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #8601

### CODE REVIEW

File: plugin-template [hunk 1]
```diff
@@ -61,7 +61,6 @@ DEFAULT_SETTINGS = {
 
 DEPRECATED_FILES = {
     "github": [
-        ".ci/scripts/cherrypick.sh",
```
Comment (daviddavis) [inline plugin-template:64]: Maybe I should give this script a new name..

Comment (mdellweg) [reply plugin-template:64]: Why, the template will replace the once deprecated one, right?

Comment (daviddavis) [reply plugin-template:64]: Yes, it should.

**Review by mdellweg — COMMENTED** [review]
Did i forget to send this?

File: templates/github/.ci/scripts/cherrypick.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+commit_message=$(git log --format=%B -n 1 $commit)
+
+if [ "$(git branch --show-current)" != "$release_branch" ]
+then
+  echo "Please checkout the $release_branch release branch first."
+  exit
+fi
+
+git checkout -b "$release_branch-cp" || git checkout "$release_branch-cp"
+git cherry-pick -n $commit
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply templates/github/.ci/scripts/cherrypick.sh.j2:21]: I had to look that one up.

File: templates/github/.ci/scripts/cherrypick.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+backport="$4"
+commit_message=$(git log --format=%B -n 1 $commit)
+
+if [ "$(git branch --show-current)" != "$release_branch" ]
+then
+  echo "Please checkout the $release_branch release branch first."
+  exit
+fi
+
+git checkout -b "$release_branch-cp" || git checkout "$release_branch-cp"
```
[the remark this answers is not in the document]
Comment (daviddavis) [reply templates/github/.ci/scripts/cherrypick.sh.j2]: I'd love to but I'm still on CentOS 7 with git versoin 1.8.3 and switch isn't available.

[the remark this answers is not in the document]
Comment (daviddavis) [reply templates/github/.ci/scripts/cherrypick.sh.j2]: I removed the checkout and now I'm assuming that the user is on the correct branch already. I added a note to the help screen about this.

**Review by mdellweg — APPROVED** [review]
Really nice tool!

**Pull request merged**